docs/dev/cancel.py

```diff
@@ -0,0 +1,46 @@
+import os
+
+import grpc
+from armada_client.client import ArmadaClient
+
+
+def cancel_jobset(client):
+    """
+    Submits and Cancels Jobs by two different methods:
+    - Cancelling a job by its job id
+    - Cancelling a job by its job set id
+    """
+
+    job_set_id = "testing"
+    queue = "default"
+
+    # This job is cancelled using the queue and job_set
+    client.cancel_jobs(queue=queue, job_set_id=job_set_id)
+
+
+def cancel():
+
+    # Ensures that the correct channel type is generated
+    if DISABLE_SSL:
```

Comment:
I see this 'correct channel type' selection/generation happening in multiple places. Can you refactor this logic into a seperate function instead of repeating it?

Reply:
Good shout. Shall do :))

Reply:
Actually in hindsight, I think I suggested having something like this as a function in the main armada-python client. The end conclusion was that this does not make much sense, as the auth will change significantly within GR clusters / different setups.

This is addressed in https://github.com/G-Research/armada-jupyter/issues/57

Reply:
There are only 2 of these in the code now, one of them is not in this module. As I explained, we didn't make this a function for a reason in python-client, and making it a function here does not make sense. Happy to move this discussion to the armada python repo if we need to, but for now, I think it should stay as is, as its only used once in the armada-jupyter code

Reply:
It is usually better to follow the pattern of using functions so you don’t replicate the same code in multiple places.  If there is a new auth variable you have to replicate that change wherever you have this code block.

@ClifHouck is suggesting the concept of DRY (Do not repeat yourself). Main benefits is too avoid someone changing one code block without changing all others. Happens a lot in large code bases

Reply:
Yes, I'm essentially advocating DRY. Better to have just one chunk of code that does task `foo` instead of `n`. If you have to change something it's better if you only have to change `func foo` instead of `n` chunks of code scattered throughout the codebase. Makes testing easier too.

Reply:
I don't think I am making it clear. I understand the concepts you are saying, and usually follow them - it just doesn't apply here.